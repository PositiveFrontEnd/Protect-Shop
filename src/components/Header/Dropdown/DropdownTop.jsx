import React from 'react';
import Close from '../HeaderSvg/close.svg?react'
import PropTypes from "prop-types";

const DropdownTop = ({ click }) => {

    return (
        <div className='dropdown__top'>
            <div></div>
            <h4 className='dropdown__title' onClick={click}>close</h4>
            <Close onClick={click} />
        </div>
    );
};

DropdownTop.propTypes = {
    click: PropTypes.func,
}

export default DropdownTop;