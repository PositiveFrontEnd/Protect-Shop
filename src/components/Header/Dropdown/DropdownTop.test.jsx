import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import DropdownTop from "./DropdownTop";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";

describe("Тестування компонента  DropdownTop", () => {
    test("Знімок", () => {
        const mockStore = configureStore();
        const initialState = {
        };
        const click = () => console.log("some text")

        const store = mockStore(initialState);
        const dropdownTop = render(
            <Provider store={store}>
                <BrowserRouter>
                    <DropdownTop click={click} />
                </BrowserRouter>
            </Provider>
        );

        expect(dropdownTop).toMatchSnapshot();
    });
});
