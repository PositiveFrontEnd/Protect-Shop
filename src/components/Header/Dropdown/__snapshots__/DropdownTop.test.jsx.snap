// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  DropdownTop > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dropdown__top"
      >
        <div />
        <h4
          class="dropdown__title"
        >
          close
        </h4>
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M7.12587 8L2 2.87413L2.87413 2L8 7.12587L13.1259 2L14 2.87413L8.87413 8L14 13.1259L13.1259 14L8 8.87413L2.87413 14L2 13.1259L7.12587 8Z"
            fill="white"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dropdown__top"
    >
      <div />
      <h4
        class="dropdown__title"
      >
        close
      </h4>
      <svg
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M7.12587 8L2 2.87413L2.87413 2L8 7.12587L13.1259 2L14 2.87413L8.87413 8L14 13.1259L13.1259 14L8 8.87413L2.87413 14L2 13.1259L7.12587 8Z"
          fill="white"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
